File: code/6.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode *p=head;
        if(p==NULL||p->next==NULL)
            return false;
        std::vector<ListNode *> Nodeaddress;
        while(p!=NULL)
        {
            if(std::find(Nodeaddress.begin(), Nodeaddress.end(), p)!=Nodeaddress.end())
                return true;
            Nodeaddress.push_back(p);
            p=p->next;
        }
        return false;
    }
};

File: code/6.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        p=head
        if(p==None or p.next==None):
            return False
        pAddress=[]
        while p!=None:
            if p not in pAddress:
                pAddress.append(p)
                p=p.next
            else:
                return True
        
        return False